app/services/CheckoutService.scala

```diff
@@ -71,21 +72,26 @@ class CheckoutService(identityService: IdentityService,
     case _ => false
   }
 
-  def processIntroductoryPeriod(defaultPaymentDelay: Days, originalCommand: Subscribe): Subscribe = {
-    val additionalRateplan = (originalCommand.ratePlans.head.productRatePlanId match {
-      case catalog.weekly.zoneA.sixWeeks.id.get => Some(catalog.weekly.zoneA.quarter)
-      case catalog.weekly.zoneC.sixWeeks.id.get => Some(catalog.weekly.zoneC.quarter)
-      case _ => None
-    }).map(plan => RatePlan(plan.id.get, None))
-
-    val updatedCommand = additionalRateplan.map { ratePlantoAdd =>
-      val updatedRatePlans = ratePlantoAdd <:: originalCommand.ratePlans
-
-      val defaultDelayDays = defaultPaymentDelay.getDays
-      val introductoryPeriodDelayDays = defaultDelayDays - 1
+  private val allSixWeekAssociations = List(catalog.weekly.zoneA.associations, catalog.weekly.zoneC.associations).flatten
+
+  // This method is not genericised because the '6' is not stored in the association.
```

Comment:
so the association will be renamed to linked or soldTogether or something, and then possibly the 6 added